larray/tests/test_axis.py

```diff
@@ -1,880 +1,439 @@
+    assert list(g[0].key) == ['a1', 'a2']
+    assert list(g[1].key) == ['a2', 'a1']
+    assert g[0].axis is alt_a
+    assert g[1].axis is alt_a
+    key = (a['a1'], a['a2'])
+    g = a[key]
+    assert isinstance(g, LGroup)
+    assert g.key == ['a1', 'a2']
+    assert g.axis is a
+    g = alt_a[key]
+    assert isinstance(g, LGroup)
+    assert g.key == ['a1', 'a2']
+    assert g.axis is alt_a
+    key = (a.i[1], a.i[2])
+    g = a[key]
+    assert isinstance(g, LGroup)
+    assert g.key == ['a1', 'a2']
+    assert g.axis is a
+    g = alt_a[key]
+    assert isinstance(g, LGroup)
+    assert g.key == ['a1', 'a2']
+    assert g.axis is alt_a
+    key = (a['a1', 'a2'], a['a2', 'a1'])
+    g = a[key]
+    assert isinstance(g, tuple)
+    assert isinstance(g[0], LGroup)
+    assert isinstance(g[1], LGroup)
+    assert g[0].key == ['a1', 'a2']
+    assert g[1].key == ['a2', 'a1']
+    assert g[0].axis is a
+    assert g[1].axis is a
+    g = alt_a[key]
+    assert isinstance(g, tuple)
+    assert isinstance(g[0], LGroup)
+    assert isinstance(g[1], LGroup)
+    assert g[0].key == ['a1', 'a2']
+    assert g[1].key == ['a2', 'a1']
+    assert g[0].axis is alt_a
+    assert g[1].axis is alt_a
+    key = (a.i[1, 2], a.i[2, 1])
+    g = a[key]
+    assert isinstance(g, tuple)
+    assert isinstance(g[0], LGroup)
+    assert isinstance(g[1], LGroup)
+    assert list(g[0].key) == ['a1', 'a2']
+    assert list(g[1].key) == ['a2', 'a1']
+    assert g[0].axis is a
+    assert g[1].axis is a
+    g = alt_a[key]
+    assert isinstance(g, tuple)
+    assert isinstance(g[0], LGroup)
+    assert isinstance(g[1], LGroup)
+    assert list(g[0].key) == ['a1', 'a2']
+    assert list(g[1].key) == ['a2', 'a1']
+    assert g[0].axis is alt_a
+    assert g[1].axis is alt_a
+
+def test_init_from_group():
+    code = Axis('code=C01..C03')
+    code_group = code[:'C02']
+    subset_axis = Axis(code_group, 'code_subset')
+    assert_array_equal(subset_axis.labels, ['C01', 'C02'])
+
+def test_matching():
+    sutcode = Axis(['A23', 'A2301', 'A25', 'A2501'], 'sutcode')
+    assert sutcode.matching('^...$') == LGroup(['A23', 'A25'])
+    assert sutcode.startingwith('A23') == LGroup(['A23', 'A2301'])
+    assert sutcode.endingwith('01') == LGroup(['A2301', 'A2501'])
+
+def test_iter():
+    sex = Axis('sex=M,F')
+    assert list(sex) == [IGroup(0, axis=sex), IGroup(1, axis=sex)]
+
+def test_positional():
+    age = Axis('age=0..115')
+    key = age.i[:-1]
+    assert key.key == slice(None, -1)
+    assert key.axis is age
+
+def test_contains():
+    # normal Axis
+    age = Axis('age=0..10')
+    age2 = age[2]
+    age2bis = age[(2,)]
+    age2ter = age[[2]]
+    age2qua = '2,'
+    age20 = LGroup('20')
+    age20bis = LGroup('20,')
+    age20ter = LGroup(['20'])
+    age20qua = '20,'
+    age247 = age['2,4,7']
+    age247bis = age[['2', '4', '7']]
+    age359 = age[['3', '5', '9']]
+    age468 = age['4,6,8'] >> 'even'
+    assert 5 in age
+    assert not ('5' in age)
```

Comment:
`'5' not in age` is slightly more readable IMO

Reply:
I used ``unittest2pytest`` tool. It works pretty well but sometimes lead to some ugly output.

Reply:
ok